default/python/tests/AI/test_system_network.py

```diff
@@ -0,0 +1,87 @@
+from unittest.mock import patch
+
+import pytest
+from AIDependencies import INVALID_ID
+
+from universe import system_network
+
+
+class TestUniverse:
+    connected_systems = {
+        1: {2, 7},
+        2: {1, 3},
+        3: {2, 4, 5, 6},
+        4: {3, 5},
+        5: {3, 4, 6},
+        6: {3, 5, 7},
+        7: {1, 6, 8},
+        8: {7, 9},
+        9: {8, 10},
+        10: {9},
+        11: {12},
+        12: {11},
+    }
+
+    # fo interface asks for an empire_id, so we need two arguments as well
+    def getImmediateNeighbors(self, sys_id, _):
+        return TestUniverse.connected_systems.get(sys_id, {})
+
+
+def test_connections():
+    """Test that the graph is bi-directional."""
+    for i1, lanes in TestUniverse.connected_systems.items():
+        for i2 in lanes:
+            assert i1 in TestUniverse.connected_systems[i2]
+
+
+def test_get_neighbors():
+    """Rather tests the patch."""
+    with patch("freeOrionAIInterface.getUniverse", new=TestUniverse):
+        assert system_network.get_neighbors(1) == {2, 7}
+
+
+@pytest.mark.parametrize(
+    ("sys_id", "expected"),
+    ((1, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}), (4, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}), (12, {11, 12}), (INVALID_ID, {})),
```

Comment:
Let's format it one case per line.

```suggestion
    (
            (1, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}),
            (4, {1, 2, 3, 4, 5, 6, 7, 8, 9, 10}),
            (12, {11, 12}),
            (INVALID_ID, {}),
    ),
```

Reply:
That's the way I wrote them, but black disagreed with you.
Well, I agree, so I've added another case just to make sure the don't fit into one line :-D

Reply:
You could use trailing coma to enforce multiline

```python
(1,2,3,)  
```
-> black -> 

```python
(
    1,
    2,
    3,
)
```

Reply:
Good to know.